src/proxy-ui-api/frontend/src/views/AddClient/FinishPage.vue

```diff
@@ -115,21 +163,66 @@ export default Vue.extend({
       this.createClient(true);
     },
 
-    generateCsr(): void {
+    generateKeyAndCsr(): void {
       const tokenId = this.$store.getters.csrTokenId;
 
-      this.$store.dispatch('generateKeyAndCsr', tokenId).then(
-        () => {
+      this.$store
+        .dispatch('generateKeyAndCsr', tokenId)
+        .then(
+          () => {
+            this.$emit('done');
+          },
+          (error) => {
+            this.$store.dispatch('showError', error);
+          },
+        )
+        .finally(() => {
           this.disableCancel = false;
           this.submitLoading = false;
-          this.$emit('done');
-        },
-        (error) => {
-          this.$store.dispatch('showError', error);
+        });
+    },
+
+    generateCsr(): void {
+      const tokenId = this.$store.getters.csrTokenId;
+
+      this.$store
+        .dispatch('generateCsr', tokenId)
+        .then(
+          () => {
+            this.$emit('done');
+          },
+          (error) => {
+            this.$store.dispatch('showError', error);
+          },
+        )
+        .finally(() => {
           this.disableCancel = false;
           this.submitLoading = false;
-        },
+        });
+    },
+
+    registerClient(): void {
+      const clientId = createClientId(
+        this.currentSecurityServer.instanceId,
```

Comment:
This doesn't seem to work, API call that results from this is "/api/v1/clients/undefined:ORG:TEST/register" (instance is undefined), resulting in failure to register the client.

I think this identifier should also include the subsystem that is being added.

Reply:
Fixed